mono/metadata/threads.c

```diff
@@ -257,8 +257,14 @@ mono_thread_get_tls_key (void)
 gint32
 mono_thread_get_tls_offset (void)
 {
-	int offset;
+	int offset = -1;
+
+#ifdef HOST_WIN32
```

Comment:
Correct, this fix followed the same pattern already in use on windows in the various mono_thread_get_xx_offset functions. I think the overall TLS support (especially on windows) will need to be worked through. The current logic around MONO_THREAD_VAR_OFFSET is implemented using keys allocated using "fast TLS" logic while windows uses a different key allocated using mono_native_tls_alloc method call... I think we could go with the above for now since it at least follows the same pattern already in use and place a "TODO" on changing the TLS implementation on Windows.